project_scrum/project_scrum.py

```diff
@@ -0,0 +1,500 @@
+            
+    @api.depends("task_ids")
+    def _task_count(self):    # method that calculate how many tasks exist
+        for p in self:
+            p.task_count = len(p.task_ids)
+            
+    @api.depends("test_ids")        
+    def _test_count(self): 
+          # method that calculate how many test cases exist
+        for p in self:
+            p.test_count = len(p.test_ids)
+
+    def _resolve_project_id_from_context(self, cr, uid, context = None):
+        """ Returns ID of project based on the value of 'default_project_id'
+            context key, or None if it cannot be resolved to a single
+            project.
+        """
+        if context is None:
+            context = {}
+        if type(context.get('default_project_id')) in (int, long):
+            return context['default_project_id']
+        if isinstance(context.get('default_project_id'), basestring):
+            project_name = context['default_project_id']
+            project_ids = self.pool.get('project.project').name_search(cr, uid, name=project_name, context=context)
+            if len(project_ids) == 1:
+                return project_ids[0][0]
+        return None
+
+    @api.model
+    def _read_group_sprint_id(self, present_ids, domain, **kwargs):
+        project_id = self._resolve_project_id_from_context()
+        sprints = self.env['project.scrum.sprint'].search([('project_id', '=', project_id)], order='sequence').name_get()
+        #sprints.sorted(key = lambda r: r.sequence)
+        return sprints, None
+    
+    _group_by_full = {
+        'sprint_ids': _read_group_sprint_id, 
+        }
+    name = fields.Char()
+
+class project_task(models.Model):
+    _inherit = "project.task"
+    _order = "sequence"
+    
+    @api.model
+    def create(self, vals):
+        vals['reference'] = self.env['ir.sequence'].get('project.Task')
+        return super(project_task, self).create(vals)
+   
+    user_id = fields.Many2one('res.users', 'Assigned to', select=True, track_visibility='onchange', default="")
+    actor_ids = fields.Many2many(comodel_name='project.scrum.actors', string='Actor')
+    sprint_id = fields.Many2one(comodel_name='project.scrum.sprint', string='Sprint', store=True)
+    us_id = fields.Many2one(comodel_name='project.scrum.us', string='User Stories')
+    date_start = fields.Datetime(string='Starting Date', required=False, default=fields.Datetime.now())
+    date_end = fields.Datetime(string='Ending Date', required=False)
+    use_scrum = fields.Boolean(related='project_id.use_scrum')
+    description = fields.Html('Description')
+    current_sprint = fields.Boolean(compute='_current_sprint', string='Current Sprint', search='_search_current_sprint')
+    moscow = fields.Selection([
+            (1, 'must'), 
+            (2, 'should'), 
+            (3, 'could'), 
+            (4, 'wont'), 
+            ('not_set', 'Not Set'), 
+            ], string='Moscow')
+    value = fields.Selection([
```

Comment:
i'd also advise to make this a function or better a model for easier customizability